src/HDWallet.cpp

```
@@ -226,11 +183,17 @@ std::string HDWallet<seedSize>::getExtendedPrivateKeyAccount(TWPurpose purpose,
 
     const auto curve = TWCoinTypeCurve(coin);
     const auto path = TW::derivationPath(coin, derivation);
-    auto derivationPath = DerivationPath({DerivationPathIndex(purpose, true), DerivationPathIndex(path.coin(), true)});
-    auto node = getNode(*this, curve, derivationPath);
-    auto fingerprintValue = fingerprint(&node, publicKeyHasher(coin));
-    hdnode_private_ckd(&node, account + 0x80000000);
-    return serialize(&node, fingerprintValue, version, false, base58Hasher(coin));
+    auto derivationPath = DerivationPath({DerivationPathIndex(purpose, true), DerivationPathIndex(path.coin(), true), DerivationPathIndex(account, true)});
+    auto node = getNode(*this, curve, derivationPath, TW::publicKeyHasher(coin));
+    if (node == nullptr) {
+        return "";
```

Let's throw an exception?